app/pages/project/home/index.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import talkClient from 'panoptes-client/lib/talk-client';
+import ProjectHomePage from './project-home';
+
+export default class ProjectHomeContainer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showWorkflowButtons: false,
+      researcherAvatar: null,
+      talkSubjects: []
+    };
+
+    this.fetchResearcherAvatar = this.fetchResearcherAvatar.bind(this);
+    this.fetchTalkSubjects = this.fetchTalkSubjects.bind(this);
+    this.showWorkflowButtons = this.showWorkflowButtons.bind(this);
+  }
+
+  componentDidMount() {
+    this.showWorkflowButtons();
+    this.fetchResearcherAvatar(this.props);
+    this.fetchTalkSubjects(this.props);
+  }
+
+  componentWillReceiveProps(nextProps, nextContext) {
+    if (this.context.user !== nextContext.user) {
+      this.showWorkflowButtons(nextProps, nextContext);
+    }
+
+    if (this.props.project !== nextProps.project) {
+      this.fetchResearcherAvatar(nextProps.props);
+    }
+  }
+
+  fetchTalkSubjects(props) {
+    talkClient.type('comments').get({ section: `project-${props.project.id}`, page_size: 10, sort: '-created_at', focus_type: 'Subject' })
+      .then((comments) => {
+        if (comments.length > 0) {
+          const subjectIds = comments.map(x => x.focus_id);
+          const uniqueSubjects = subjectIds.filter((el, i, arr) => { return arr.indexOf(el) === i; });
+          uniqueSubjects.splice(3, 7);
+          apiClient.type('subjects').get(uniqueSubjects)
+            .then((subjects) => {
+              this.setState({ talkSubjects: subjects });
+            });
+        }
+      }).catch(error => console.error(error));
+  }
+
+  fetchResearcherAvatar(props) {
+    if (props.project.configuration && props.project.configuration.researcherID) {
+      if (props.project.configuration.researcherID === props.project.display_name) {
+        if (props.projectAvatar && props.projectAvatar.src) {
+          this.setState({ researcherAvatar: props.projectAvatar.src });
+        }
+      }
+
+      apiClient.type('users').get(this.props.project.configuration.researcherID)
```

Comment:
Perhaps move this API call into the conditional to only occur if `props.projectAvatar` is not found? It looks like this call is being made even if the `researcherAvatar` state is being set above to a project avatar.